app/src/main/java/sensors_in_paradise/sonar/page2/RecordingDataManager.kt

```diff
@@ -97,6 +103,82 @@ class RecordingDataManager(jsonFile: File, val recordingsDir: File) : JSONStorag
         return false
     }
 
+    private fun getAbsoluteLineNumber(file: File): Int {
+        var lineNumber = 0
+
+        val reader = BufferedReader(FileReader(file))
+        while (reader.readLine() != null) lineNumber++
+        reader.close()
+
+        return lineNumber
+    }
+
+    private fun getTimeStampAtLine(file: File, lineNumber: Int): String {
+        var counter = 0
+
+        val reader = BufferedReader(FileReader(file))
+        var line: String
+        while (true) {
+            line = reader.readLine() ?: break
+
+            if (counter == lineNumber) {
+                val columns = line.split(",")
+                reader.close()
+                return columns[1]
+            }
+
+            counter++
+        }
+
+        reader.close()
+        return ""
+    }
+
+    private fun findTimeStamp(file: File, timestamp: String): Boolean {
+        var headerSize = 9
+
+        val reader = BufferedReader(FileReader(file))
+        var line: String
+        while (true) {
+            line = reader.readLine() ?: break
+
+            if (headerSize <= 0) {
+                val columns = line.split(",")
+                if (columns[1] == timestamp) {
+                    reader.close()
+                    return true
+                }
+            }
+            headerSize--
```

Comment:
I feel like it is more clean to count down in an else block than having it count to negative infinitive. Of course we would never hit the lowest value in a real world scenario but to me this is not clean and stopping it at 0 is totally fine.